salt/config/__init__.py

```diff
@@ -3649,6 +3653,21 @@ def get_id(opts, cache_minion_id=False):
     if opts.get('minion_id_lowercase'):
         newid = newid.lower()
         log.debug('Changed minion id %s to lowercase.', newid)
+
+    # Optionally remove a domain or many in a generated minion id
+    if opts.get('minion_id_remove_domain'):
+        rem_domain = opts.get('minion_id_remove_domain')
```

Comment:
Better is either:
```
￼    remove_domain = opts.get('minion_id_remove_domain')
    if remove_domain:
```
or
```
    if opts.get('minion_id_remove_domain'):
￼         remove_domain = opts.get['minion_id_remove_domain']
```

Reply:
I use your second proposal. 

I also had to move the code into a separate function because it increased `get_id()` to a  cyclomatic complexity of 22.